### DISCUSSION TITLE
Allow users to disable provided configurations for complete customization options

### DISCUSSION THREAD

**Discussion by jaydrogers** [serversideup/docker-php#173] [Ideas & Feature Requests]
# 👉 Describe the problem
[comment]: <> (What are you trying to solve?)
* People who want to have full customization of the image need to run strange and hacky scripts in order to start customizing the image

# 👥 Problem evidence & reach
[comment]: <> (How many people have this problem?)

Here is an example from a Discord user. If they want full customization of the image, then they have to run a script to delete our configs, then place their own.

<img width="688" alt="image" src="https://user-images.githubusercontent.com/3174134/230955786-5d2dccfc-944f-4262-a6be-1f4c0104885d.png">

# 🏆 How to solve this problem
[comment]: <> (Describe the feature that you are proposing and how it will solve the problem)

### Create an environment variable do disable the enabling of our default server configurations
* Create an environment variable like `DISABLE_DEFAULT_CONFIG`
* If `true`, then: Do not enable the server configurations we provide. The user will be required to set their own server configuration

# 💯 How do we validate the problem is solved?
[comment]: <> (Explain use cases on how we can measure the success of this implementation)
- Users will not need to run hacky scripts to delete server configurations
- Users can easily place their own configurations to have full control of the web server

**Comment by sellrock-de** [comment]
i copy my own http.conf in the sites enabled, but i need to do a restart after that. Its not good

**Reply by jaydrogers** [reply]
Just an FYI, to provide your own configuration would would run a Docker build and just copy over my files. This would put your configurations in place without needing a restart.

**Comment by jaydrogers** [comment]
This has been launched in our V3 beta https://release-v3-0.docker-php.pages.dev/docs/guide/adding-your-own-start-up-scripts
